src/shotgunEventDaemon.py

```diff
@@ -342,10 +342,40 @@ def _loadEventIdData(self):
                     # Provide event id info to the plugin collections. Once
                     # they've figured out what to do with it, ask them for their
                     # last processed id.
+                    noStateCollections = []
                     for collection in self._pluginCollections:
                         state = self._eventIdData.get(collection.path)
                         if state:
                             collection.setState(state)
+                        else:
+                            noStateCollections.append(collection)
+
+                    # If we don't have a state it means there's no match
+                    # in the id file. First we'll search to see the latest id a
+                    # matching plugin name has elsewhere in the id file. We do
+                    # this as a fallback in case the plugins directory has been
+                    # moved. If there's no match, use the latest event id 
+                    # in Shotgun.
+                    if noStateCollections:
+                        maxPluginStates = {}
+                        for collection in self._eventIdData.values():
+                            for pluginName, pluginState in collection.items():
+                                if pluginName in maxPluginStates.keys():
+                                    if pluginState[0] > maxPluginStates[pluginName][0]:
+                                        maxPluginStates[pluginName] = pluginState
+                                else:
+                                    maxPluginStates[pluginName] = pluginState
+
+                        for collection in noStateCollections:
+                            state = collection.getState()
+                            for pluginName in state.keys():
+                                if pluginName in maxPluginStates.keys():
+                                    state[pluginName] = maxPluginStates[pluginName]
+                                else:
+                                    lastEventId = self._getLastEventIdFromDatabase()
```

Comment:
I feel like this should be called once if necessary and the value saved for all required usages in the loop instead of calling it multiple times and potentially have the lastEventId change as you move forward through the loop (for sites that generate lots of events). Thoughts?